hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -101,11 +101,7 @@ protected BaseTableMetadata(HoodieEngineContext engineContext, HoodieMetadataCon
       try {
         return fetchAllPartitionPaths();
       } catch (Exception e) {
-        if (metadataConfig.enableFallback()) {
-          LOG.error("Failed to retrieve list of partition from metadata", e);
-        } else {
-          throw new HoodieMetadataException("Failed to retrieve list of partition from metadata", e);
-        }
+        throw new HoodieMetadataException("Failed to retrieve list of partition from metadata", e);
```

Comment:
remove enable fallback and throw the exception here is to make the metadata table sync fast fail?

Reply:
To make lookups from metadata table fail fast. 

When fallback was enabled, any errors in lookups from metadata table are logged and file-listing from file system is used instead.

Reply:
@prashantwason So when using metadata table and found an error, what's the practice to handle the error or how to  recover? just disable metadata table?

Reply:
Yes, disabling metadata table is the right approach for a quick fix. There is no user-fixable part of the table.

In the long term, we expect the table to be very stable for the use-cases it handles.